common/djangoapps/third_party_auth/models.py

```diff
@@ -241,22 +241,31 @@ def get_register_form_data(cls, pipeline_kwargs):
             values for that field. Where there is no value, the empty string
             must be used.
         """
+        current_provider = kwargs.pop('current_provider', None)
+        registration_form_data = {}
+
         # Details about the user sent back from the provider.
         details = pipeline_kwargs.get('details')
+        registration_form_data['email'] = details.get('email', '')
+        registration_form_data['name'] = details.get('fullname', '')
+
+        # There are often nonstandard requirements for some registration flows,
+        # like the user's country code.
+        if current_provider.required_extra_attributes:
+            attributes = pipeline_kwargs['response']['attributes']
+            for attr, val in current_provider.required_extra_attributes.iteritems():
+                if val:
+                    registration_form_data[attr] = attributes[val]
```

Comment:
@bradenmacdonald Well for one, it wouldn't be coming in `details` because that's something PSA fills for us and PSA isn't filling up country data.

In that case one would have to gather this data directly from `pipeline_kwargs['response']['attributes']` the way I attempt here above. But not all backends get their data in this way, so this wouldn't work as an unconditional, general functionality.